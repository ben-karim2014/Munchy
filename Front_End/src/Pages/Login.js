//This is a public page route
import { Redirect, useHistory } from "react-router-dom"
import React, {Component, Fragment} from 'react';
import HeaderApp from '../Components/Header'
import { faUserAlt, faLock} from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {login} from '../actions/authAction'
import {clearErrors} from '../actions/errorActions'
import {
    Button,
    Form,
    FormGroup,
    Label,
    Input,
    Row,
    Col,
    Alert,
    Container,
    Card,
    CardBody,
    CardTitle,
    CardText,
    InputGroup,
    CardLink,



} from 'reactstrap'


import {connect} from 'react-redux'
import PropTypes from 'prop-types'

class Login extends Component{
    state= {
       
        email: '',
        password: '',
        msg:null
        
    }
    static propTypes = {
        isAthenticated : PropTypes.bool,
        error : PropTypes.object.isRequired,
        login: PropTypes.func.isRequired,
        clearErrors: PropTypes.func.isRequired
    }
    componentDidUpdate(prevProps){
        const {error} = this.props;
        
        if(error !== prevProps.error){
            if(error.id=== 'LOGIN_FAIL'){
                this.setState({msg: error.msg})

            }
            else {
                this.setState({ msg: null})
            }
        }

    }
    
    onSubmit = e =>{
        this.props.clearErrors();
        e.preventDefault();
        const {email, password} = this.state;

        //Create User Object
        const newUser =
        {
           
            email,
            password,
          
        }
        //Attempt to login
        
            this.props.login(newUser);
          //  window.location.reload(true);
       
        
       // console.log(`The message is : ${this.state.msg}`);

    }
    changeValue = (e) => {
        this.props.clearErrors();
        this.setState({
         [e.target.name] : e.target.value   })
    }
    render(){
        const {isAuthenticated, user} = this.props.auth;
        return (
            
            <Fragment>
            <HeaderApp />
            {isAuthenticated ? <Redirect to={{ pathname:"/home", state: { isAuthenticated: isAuthenticated, user: user }}}/> : null}
            {this.state.msg ? (<Alert color="danger">{this.state.msg}</Alert>) : null}
            <Form  onSubmit={this.onSubmit}>
            
            <Container fluid className="full-height bg-light">
            <Row className="h-100 justify-content-center full-height align-items-center Login-wrap">
              <Col xs="10" lg="3" className="p-0">
                <Card>
                  <CardBody>
                    <CardTitle>LOGIN</CardTitle>
                    <CardText>Sign in with your account.</CardText>
                    <InputGroup className="mb-3">
                      <div className="input-group-prepend">
                        <span className="input-group-text">
                        <FontAwesomeIcon icon={faUserAlt} />
                        </span>
                      </div>
                      <Input type="text" name='email' id='email' placeholder="Email" onChange={this.changeValue}/>
                    </InputGroup>
                    <InputGroup className="mb-4">
                      <div className="input-group-prepend">
                        <span className="input-group-text">
                        <FontAwesomeIcon icon={faLock} />
                        </span>
                      </div>
                      <Input type="password" name='password' id='password' placeholder="Password" onChange={this.changeValue}/>
                    </InputGroup>
                    <Row>
                      <Col xs="12" lg="6">
                        <Button color="primary" className="px-4">Login</Button>
                      </Col>
                   
                    </Row>
                  </CardBody>
                </Card>
              </Col>
              
            
            </Row>
             
           <Row className="h-100 justify-content-center full-height align-items-center Login-wrap">
           <Col xs="10" lg="3" className="p-0">
           <Card color="dark">
             <CardBody className="text-white">
               <CardTitle>CREATE ACCOUNT</CardTitle>
               <CardText>Don't have an account. Please create an account </CardText>
            
               <Col xs="12" lg="6" className="text-right">
                   <Button color="link" className="px-0" href="/signup"> >>Join</Button>
                 </Col>
               
             </CardBody>
           </Card>
          
           </Col>
           </Row>
          </Container>
            </Form>
            </Fragment>
          );
    }

}
const mapStateToProps = state =>({
    isAthenticated : state.auth.isAthenticated,
    error: state.error,
    auth: state.auth

})
export default connect (
    mapStateToProps,
    {login, clearErrors}
)(Login);
