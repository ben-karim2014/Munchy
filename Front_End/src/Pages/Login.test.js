import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from "react-redux";
import store from "../store";

import Login from './Login';

it('renders correctly', () => {
    const tree = renderer.create(<Provider store={store}><Login /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});