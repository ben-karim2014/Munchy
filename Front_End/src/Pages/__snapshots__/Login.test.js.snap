// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="bottomBorder"
    data-testid="upper"
  >
    <nav
      className="header navbar navbar-expand-sm navbar-dark"
      data-testid="upper2"
    >
      <a
        href="/"
      >
        <img
          alt=""
          className="logo"
          src="munchy_logo.png"
        />
      </a>
      <button
        aria-label="Toggle navigation"
        className="navbar-toggler"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        data-testid="upper3"
        style={Object {}}
      >
        <div
          className="header"
        >
          <ul
            className="navbar-header center navbar-nav"
          >
            <li
              className="nav-list nav-item"
            >
              <a
                className="nav-link"
                href="/"
                onClick={[Function]}
              >
                Home
              </a>
            </li>
            <li
              className="nav-list nav-item"
            >
              <a
                className="nav-link"
                href="/login"
                onClick={[Function]}
              >
                Sign in
              </a>
            </li>
            <li
              className="nav-list nav-item"
            >
              <a
                className="nav-link"
                href="/signup"
                onClick={[Function]}
              >
                Sign up
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>,
  <div
    className="signin-form"
  >
    <form
      className=""
      onSubmit={[Function]}
    >
      <div
        className="full-height bg-light container-fluid"
      >
        <div
          className="h-100 justify-content-center full-height align-items-center Login-wrap row"
        >
          <div
            className="p-0 col-10 col-lg-3"
          >
            <div
              className="card"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-sign-in-alt fa-w-16 fa-2x "
                    data-icon="sign-in-alt"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M416 448h-84c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h84c17.7 0 32-14.3 32-32V160c0-17.7-14.3-32-32-32h-84c-6.6 0-12-5.4-12-12V76c0-6.6 5.4-12 12-12h84c53 0 96 43 96 96v192c0 53-43 96-96 96zm-47-201L201 79c-15-15-41-4.5-41 17v96H24c-13.3 0-24 10.7-24 24v96c0 13.3 10.7 24 24 24h136v96c0 21.5 26 32 41 17l168-168c9.3-9.4 9.3-24.6 0-34z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                     SIGN IN
                </div>
                <hr />
                <p
                  className="card-text"
                >
                  Sign in with your account.
                </p>
                <div
                  className="mb-3 input-group"
                >
                  <div
                    className="input-group-prepend"
                  >
                    <span
                      className="input-group-text"
                    >
                      <svg
                        aria-hidden="true"
                        className="svg-inline--fa fa-user-alt fa-w-16 "
                        data-icon="user-alt"
                        data-prefix="fas"
                        focusable="false"
                        role="img"
                        style={Object {}}
                        viewBox="0 0 512 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M256 288c79.5 0 144-64.5 144-144S335.5 0 256 0 112 64.5 112 144s64.5 144 144 144zm128 32h-55.1c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16H128C57.3 320 0 377.3 0 448v16c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48v-16c0-70.7-57.3-128-128-128z"
                          fill="currentColor"
                          style={Object {}}
                        />
                      </svg>
                    </span>
                  </div>
                  <input
                    className="form-control"
                    id="email"
                    name="email"
                    onChange={[Function]}
                    placeholder="Email"
                    type="text"
                  />
                </div>
                <div
                  className="mb-4 input-group"
                >
                  <div
                    className="input-group-prepend"
                  >
                    <span
                      className="input-group-text"
                    >
                      <svg
                        aria-hidden="true"
                        className="svg-inline--fa fa-lock fa-w-14 "
                        data-icon="lock"
                        data-prefix="fas"
                        focusable="false"
                        role="img"
                        style={Object {}}
                        viewBox="0 0 448 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z"
                          fill="currentColor"
                          style={Object {}}
                        />
                      </svg>
                    </span>
                  </div>
                  <input
                    className="form-control"
                    id="password"
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                  />
                </div>
                <div
                  className="row"
                >
                  <div
                    className="col-12 col-lg-6"
                  >
                    <button
                      aria-label={null}
                      className="px-4 btn btn-secondary"
                      id="click-login"
                      onClick={[Function]}
                    >
                      Login
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="h-100 justify-content-center full-height align-items-center Login-wrap row"
        >
          <div
            className="p-0 col-10 col-lg-3"
          >
            <div
              className="card bg-dark"
            >
              <div
                className="text-white card-body"
              >
                <div
                  className="text-center col-12 col-lg-12"
                >
                  <a
                    className="link"
                    href="/signup"
                  >
                     Or Just Sign Up
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>,
]
`;
